File: app/src/main/java/com/hqq/core/app/ui/activity/bar/SearchBarActivity.java
package com.hqq.core.app.ui.activity.bar;

import android.app.Activity;
import android.content.Intent;

import com.hqq.core.app.R;
import com.hqq.core.app.weight.BaseToolBarSearch;
import com.hqq.core.ui.BaseActivity;

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.iblibrary.ui.activity
 * @FileName :   SearchBarActivity
 * @Date : 2018/12/14 0014
 * @Descrive : TODO
 * @Email :  qiqiang213@gmail.com
 */
public class SearchBarActivity extends BaseActivity {
    public static void open(Activity context) {
        Intent starter = new Intent(context, SearchBarActivity.class);
        context.startActivityForResult(starter, -1);
    }

    @Override
    public int setViewId() {
        return R.layout.activity_search_bar;
    }

    @Override
    public void initDefConfig() {
        super.initDefConfig();
        mRootViewBuild.setIToolBarClass(BaseToolBarSearch.class);
    }

    @Override
    public void initView() {

    }

}

File: app/src/main/java/com/easy/example/ui/bar/SearchBarActivity.kt
package com.easy.example.ui.bar

import android.app.Activity
import android.content.Intent
import com.easy.core.ui.base.BaseActivity
import com.easy.example.R
import com.easy.example.widget.BaseToolBarSearch

/**
 * @Author : huangqiqiang
 * @Package : com.easy.iblibrary.ui.activity
 * @FileName :   SearchBarActivity
 * @Date : 2018/12/14 0014
 * @Descrive : TODO
 * @Email :  qiqiang213@gmail.com
 */
class SearchBarActivity : BaseActivity() {


    override fun initConfig() {
        super.initConfig()

        rootViewImpl.iToolBarBuilder.iCreateToolbar = BaseToolBarSearch()


    }

    override fun getLayoutViewId(): Int {
        return  R.layout.activity_search_bar
    }

    override fun initView() {}

    companion object {
        fun open(context: Activity) {
            val starter = Intent(context, SearchBarActivity::class.java)
            context.startActivityForResult(starter, -1)
        }
    }
}